File: leetcode/0066-plus-one/solution.py
class Solution(object):
    def plusOne(self, digits):
        if digits[-1] != 9:
            digits[-1] = digits[-1] + 1
        elif len(digits)==1:
            digits=[1,0]
        else:
            for i in range(len(digits)-1,-1,-1):
                if digits[i]==9 and i==0:
                    digits[i]=1
                    digits.append(0)
                elif digits[i]==9:
                    digits[i]=0
                else:
                    digits[i]=digits[i]+1
                    break
        return digits

        """
        :type digits: List[int]
        :rtype: List[int]
        """
        

File: leetcode/0066-plus-one/solution.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        auto n=digits.end() - 1;
        int rem=1;
        do{
            int sum= *n+rem;
            *n=sum%10;
            rem=sum/10;
            n=n-1;
        }while(rem==1 && n>=digits.begin());
        if (rem==1){
            digits.insert(digits.begin(),rem);
        }
        return digits;
    }
};
